### ISSUE TITLE
Help with including terrain data in Fluidity for 3D flow analysis on a mountain surface

### ISSUE THREAD

**Issue by gbhutani** [FluidityProject/fluidity#261]
Hello,

I need help with including terrain geometry in Fluidity for 3D flow analysis on a mountain topography. Has anyone imported (x,y,z) data from Google Earth and converted that to a 3D volume? Any help will be useful in this direction. Is there any meshing software that is able to provide advanced meshing features (through GUI) and produce Gmsh format meshes (.msh) that can be imported in Fluidity?

**Comment by Patol75** [comment]
I have never done that before, but if I had to, I guess I would try my luck with Python and [meshio](https://pypi.org/project/meshio/). If you have the coordinates of each point, you can generate the Triangulation with [scipy.spatial.Delaunay](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.Delaunay.html) and, hopefully, provide it to a meshio primitive.

Another existing library is [PyMesh](https://pymesh.readthedocs.io/en/latest/).

**Comment by GNikit** [comment]
You can also use [Cubit](https://cubit.sandia.gov/) (if you are in the US and have a license) or [Trelis](https://csimsoft.com/trelis.jsp) (license can be bought although a bit expensive) to import/design you mesh and then export it to NASTRAN format (.bdf)  that gmsh can handle, then you only need to tag the physical boundaries, see http://onelab.info/pipermail/gmsh/2008/003222.html. To be perfectly clear, this will be the triangulated mesh, not the CAD file. I am sure there is a way to convert the CAD file into some format and then import it to gmsh but I haven't looked into it.

Having said that, you don't really need Trelis or Cubit, you can use any CADing software that can export to (.bdf) and then import it to gmsh.

There is also some very limited support for `exodus2` and `vtu` meshes, although the former only works in serial and the latter cannot have any physical IDs from what I understand.

I hope that helps.

**Comment by gbhutani** [comment]
Dear Thomas and GNikit, Thanks a lot for your comments. We tried a few options (mostly oprn source) and were able to successfully generate the model (and mesh it) in GMSH. We imported an initial .stl from Google Earth Pro, saved it as an .igs using ANSYS Workbench (requires license), imported the .igs in GMSH and saved it as a .geo. Finally added Physical Boundaries in the .geo file in the text mode. A .gif of the final simulation is attached. 
![animation0-10](https://user-images.githubusercontent.com/7738499/79308598-7f4d4580-7f16-11ea-98cf-35fadf8fc583.gif)
